bundles/mapping/mapmodule/oskariStyle/generator.ol.js

```diff
@@ -1,15 +1,223 @@
-
 import olStyleStyle from 'ol/style/Style';
 import olStyleFill from 'ol/style/Fill';
 import olStyleStroke from 'ol/style/Stroke';
 import olStyleCircle from 'ol/style/Circle';
 import olStyleIcon from 'ol/style/Icon';
 import olStyleText from 'ol/style/Text';
 
-import { LINE_DASH, LINE_JOIN, EFFECT, FILL_STYLE } from './constants';
+import { LINE_DASH, LINE_JOIN, EFFECT, FILL_STYLE, STYLE_TYPE } from './constants';
+import { filterOptionalStyle, getOptionalStyleFilter } from './filter';
 
+const log = Oskari.log('MapModule.util.style');
 const TRANSPARENT = 'rgba(0,0,0,0)';
 
+const merge = (...styles) => jQuery.extend(true, {}, ...styles);
+
+export const HIDDEN_STYLE = new olStyleStyle({ visibility: 'hidden' });
+
+const defaultStyles = {};
+export const setDefaultStyle = (layerType, styleDef) => {
+    defaultStyles[layerType] = styleDef;
+};
+const getDefaultStyle = layerType => {
+    return defaultStyles[layerType] || {};
+};
+
+const getFeatureStyle = (layer, extendedDef = {}) => {
+    const style = layer.getCurrentStyle();
+    const defaultDef = getDefaultStyle(layer.getLayerType());
+    return merge(defaultDef, style.getFeatureStyle(), extendedDef);
+};
+
+export const useStyleFunction = layer => {
+    const styleType = geometryTypeToStyleType(layer.getGeometryType());
+    const hasOptionalStyles = layer.getCurrentStyle().getOptionalStyles().length > 0;
+    const hasCluster = typeof layer.getClusteringDistance() !== 'undefined';
+    return hasOptionalStyles || hasCluster ||
+        !styleType || styleType === STYLE_TYPE.COLLECTION;
+};
+
+export const geometryTypeToStyleType = type => {
+    let styleType;
+    switch (type) {
+    case 'LineString':
+    case 'MultiLineString':
+        styleType = STYLE_TYPE.LINE; break;
+    case 'Polygon':
+    case 'MultiPolygon':
+        styleType = STYLE_TYPE.AREA; break;
+    case 'Point':
+    case 'MultiPoint':
+        styleType = STYLE_TYPE.POINT; break;
+    case 'GeometryCollection':
+        styleType = STYLE_TYPE.COLLECTION; break;
+    }
+    return styleType;
+};
+
+export const getOlStyleForLayer = (mapmodule, layer, extendedDef) => {
+    const featureStyle = getFeatureStyle(layer, extendedDef);
+    if (!useStyleFunction(layer)) {
+        const styleType = geometryTypeToStyleType(layer.getGeometryType());
+        return mapmodule.getStyle(featureStyle, styleType);
+    }
+    const typed = mapmodule.getGeomTypedStyles(featureStyle);
+    const optional = layer.getCurrentStyle().getOptionalStyles().map(optionalDef => {
+        return {
+            filter: getOptionalStyleFilter(optionalDef),
+            typed: mapmodule.getGeomTypedStyles(merge(featureStyle, optionalDef))
+        };
+    });
+    return getStyleFunction({ typed, optional });
+};
+
+export const getStyleForGeometry = (geometry, styleTypes) => {
```

Comment:
Do all of these need to be exported? If not remove exports so it's easier to read what functions are helpers for this file vs used from outside.